### PR TITLE
Feature/uppsf 3344 migrate to msk

### PR DESCRIPTION
**Pull request by martin-stanchev**
# Description

## What
Part of MSK migration

## Why

https://financialtimes.atlassian.net/browse/UPPSF-3344

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: health/healthcheck_test.go [hunk 1]
```diff
@@ -21,12 +21,14 @@ func (mc mockConsumer) ConnectivityCheck() error {
 	return mc.err
 }
 
-func (mc mockConsumer) StartListening(messageHandler func(message kafka.FTMessage) error) {
-	return
+func (mc mockConsumer) MonitorCheck() error {
```
Comment (atanasdinov) [inline health/healthcheck_test.go:23]: Let's please include this check in the tests.

File: health/healthcheck_test.go [hunk 2]
```diff
@@ -21,12 +21,14 @@ func (mc mockConsumer) ConnectivityCheck() error {
...
+	return mc.err
+}
+
+func (mc mockConsumer) Start(messageHandler func(message kafka.FTMessage)) {
```
Comment (atanasdinov) [inline health/healthcheck_test.go]: With the introduction of the local interfaces all other functions i.e. `Start(), SendMessage(), Close()` etc. can be removed.

File: helm/pac-annotations-mapper/app-configs/pac-annotations-mapper_eks_delivery.yaml [hunk 3]
```diff
@@ -6,3 +6,5 @@ env:
   CONSUMER_GROUP: pac-annotations-mapper
   CONSUMER_TOPIC: NativeCmsMetadataPublicationEvents
   PRODUCER_TOPIC: ConceptAnnotations
+  KAFKA_LAG_TOLERANCE: 200
+  LOG_LEVEL: "INFO"
```
Comment (atanasdinov) [inline helm/pac-annotations-mapper/app-configs/pac-annotations-mapper_eks_delivery.yaml:10]: We'd need to add this variable to the `Deployment` config as well.

**Review by dtvalk-ov — DISMISSED** [review]

File: health/healthcheck.go [hunk 4]
```diff
@@ -87,6 +95,18 @@ func (h *HealthCheck) writeQueueCheck() fthealth.Check {
 	}
 }
 
+func (h *HealthCheck) kafkaConsumerMonitoringCheck() fthealth.Check {
+	return fthealth.Check{
+		ID:               "kafka-consumer-lag-check",
+		Name:             "Check Kafka consumer status",
+		Severity:         3,
+		BusinessImpact:   "Consumer is lagging behind when reading messages. Ingestion is delayed.",
+		TechnicalSummary: "Messages awaiting handling exceed the configured lag tolerance. Check if Kafka consumer is stuck.",
```
Comment (ilian2233) [inline health/healthcheck.go]: **nitpick(non-blocking):** The [constant provided by kafka-client-go](https://github.com/Financial-Times/kafka-client-go/blob/v3/consumer.go#L26) can be used here

File: service/messageHandler.go [hunk 5]
```diff
@@ -26,41 +27,48 @@ var predicates = map[string]string{
 	"http://www.ft.com/ontology/annotation/mentions":           "mentions",
 }
 
+type kafkaProducer interface {
+	SendMessage(message kafka.FTMessage) error
```
Comment (atanasdinov) [inline service/messageHandler.go:31]: Only this function is necessary for the definition of the interface. The others can be removed.

File: service/messageHandler_test.go [hunk 6]
```diff
@@ -213,32 +204,23 @@ func TestMessageProducerError(t *testing.T) {
...
 		Body:    `{"foo":"bar"}`,
 	}
 
-	actual := service.HandleMessage(inbound)
-	assert.NoError(t, actual)
+	service.HandleMessage(inbound)
 	mp.AssertExpectations(t)
 
-	actualLog := hook.LastEntry()
-	assert.Equal(t, "Skipping this message because the whitelist is invalid.", actualLog.Message, "log message")
```
Comment (atanasdinov) [inline service/messageHandler_test.go:242]: Just wondering if there's something concerning about the hook assertions that we've removed?

**Review by atanasdinov — COMMENTED** [review]
The changes are great! Most of these comments are just nitpicks for clearer code but could we please set the `replicaCount` to 1 both in `values.yaml` and in the app config?

**Review by ilian2233 — DISMISSED** [review]
Looks good to me.

File: service/messageHandler_test.go (hunk 6 above)
Comment (martin-stanchev) [reply service/messageHandler_test.go:242]: Well they rely on the old logic of the logger where we don't pass a pointer to a logger instance, but instead the logger is initialised in main.go and used everywhere directly from the library, so `hook.LastEntry()` will return nil on all the tests.

**Review by atanasdinov — DISMISSED** [review]

**Review by dtvalk-ov — DISMISSED** [review]

**Review by ilian2233 — APPROVED** [review]

**Review by dtvalk-ov — APPROVED** [review]

**Pull request merged**